corehq/apps/users/models.py

```diff
@@ -1629,14 +1629,20 @@ def get_owner_ids(self):
     def retire(self):
         suffix = DELETED_SUFFIX
         deletion_id = random_hex()
+        deleted_cases = set()
         # doc_type remains the same, since the views use base_doc instead
         if not self.base_doc.endswith(suffix):
             self.base_doc += suffix
             self['-deletion_id'] = deletion_id
-        for formlist in chunked(self.get_forms(wrap=False, include_docs=True), 50):
-            tag_docs_as_deleted.delay(XFormInstance, formlist, deletion_id)
+
         for caselist in chunked(self.get_cases(wrap=False), 50):
             tag_docs_as_deleted.delay(CommCareCase, caselist, deletion_id)
+            for case in caselist:
+                deleted_cases.add(case['_id'])
+
+        for formlist in chunked(self.get_forms(wrap=False, include_docs=True), 50):
+            tag_forms_as_deleted_rebuild_associated_cases.delay(XFormInstance, formlist, deletion_id,
```

Comment:
Seems unnecessary to pass in the XFormInstance class here since this will only ever get called for xforms.